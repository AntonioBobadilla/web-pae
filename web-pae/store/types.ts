import { Subject } from '@/components/search-bar';

export const days: {
  [key: string]: number;
} = {
  lunes: 0,
  martes: 1,
  miercoles: 2,
  jueves: 3,
  viernes: 4,
  sabado: 5
};

export interface Period {
  dia: string;
  inicio: number;
  fin: number;
  id: string;

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
}

// Define a type for the slice state
export interface CreateTutorState {
  name: string;
  email: string;
  major: string;
  password: string;
  passwordConfirmation: string;
  schedule: {
    firstPeriod: Period[];
    secondPeriod: Period[];
    thirdPeriod: Period[];
  };
  subjects: Subject[];
  isLoading: boolean;
  error: string;
}
