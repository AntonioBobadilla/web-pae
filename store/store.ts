import { configureStore } from '@reduxjs/toolkit';
import { createWrapper } from 'next-redux-wrapper';
import { Action } from 'redux';
import thunk, { ThunkAction } from 'redux-thunk';
import createTutor from './reducers/create-tutor';
import scheduleTutoring from './reducers/schedule-tutoring';
import user from './reducers/user';

const saveToLocalStorage = (state: RootState) => {
  try {
    localStorage.setItem('state', JSON.stringify(state));
  } catch (e) {
    // console.error(e);
  }
};

const loadFromLocalStorage = () => {
  try {
    const stateStr = localStorage.getItem('state');
    return stateStr ? JSON.parse(stateStr) : undefined;
  } catch (e) {
    // console.error(e);
    return undefined;
  }
};

const persistedStore = loadFromLocalStorage();

// middleware
const middleware = [thunk];

// creating store

export const store = configureStore({
  reducer: {
    user,
    scheduleTutoring,
    createTutor
  },
  middleware,
  preloadedState: persistedStore
});

store.subscribe(() => {
  saveToLocalStorage(store.getState());
});

// assigning store to next wrapper
export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;

const makeStore = () => store;

export const wrapper = createWrapper(makeStore);
