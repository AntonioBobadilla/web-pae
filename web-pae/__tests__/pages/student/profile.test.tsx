import Profile from '@/pages/student/profile';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { store } from 'store/store';

describe('Student Profile', () => {
  it('renders admin profile page unchanged', () => {
    const { container } = render(
      <Provider store={store}>
        <Profile />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
