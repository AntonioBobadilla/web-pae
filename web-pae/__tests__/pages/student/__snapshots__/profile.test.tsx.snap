// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student Profile renders admin profile page unchanged 1`] = `
<div>
  <div
    class="content"
  >
    <div
      class="main"
    >
      <input
        class="toggler"
        type="checkbox"
      />
      <div
        class="buttonContainer"
        id="toggle"
      >
        <img
          class="icon"
          src="/icons/gear.svg"
        />
      </div>
      <div
        class="overlay"
        id="overlay"
      >
        <nav
          class="overlayMenu"
        >
          <ul
            class="ul"
          >
            <li
              class="li"
            >
              <a
                class="a"
              >
                Modificar contraseña
              </a>
            </li>
            <li
              class="li"
            >
              <a
                class="a"
              >
                Modificar idioma
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
    <h1>
      Profile
    </h1>
    <div
      class="cardInfo"
    >
      <div
        class="box"
      >
        <div
          class="dateSize"
        >
          <h1
            class="bi bi-alarm icon"
          >
             
          </h1>
          <span
            class="caption"
          >
             
            Lunes 27 de Marzo - 3:00 PM - 4:00 PM
             
          </span>
        </div>
        <div
          class="subjectSize"
        >
          <h1
            class="bi bi-stack icon"
          >
             
          </h1>
          <span
            class="caption"
          >
             
            Estructuras simbólicas en la imagen, la literatura y la música
             
          </span>
        </div>
        <div
          class="studentSize"
        >
          <h1
            class="bi bi-person-circle icon"
          >
             
          </h1>
          <span
            class="caption"
          >
             
            Karen Rugerio Armenta
             
          </span>
        </div>
        <div
          class="locationSize"
        >
          <a
            class="caption2"
            href="https://itesm.zoom.us/j/5519309007"
          >
             
            https://itesm.zoom.us/j/5519309007
             
          </a>
        </div>
        <div
          class="statusSize"
        >
          <button
            class="container pending"
          >
             
            Pendiente
          </button>
        </div>
      </div>
      <div
        class="box"
      >
        <div
          class="dateSize"
        >
          <h1
            class="bi bi-alarm icon"
          >
             
          </h1>
          <span
            class="caption"
          >
             
            Lunes 27 de Marzo - 3:00 PM - 4:00 PM
             
          </span>
        </div>
        <div
          class="subjectSize"
        >
          <h1
            class="bi bi-stack icon"
          >
             
          </h1>
          <span
            class="caption"
          >
             
            Estructuras simbólicas en la imagen, la literatura y la música
             
          </span>
        </div>
        <div
          class="studentSize"
        >
          <h1
            class="bi bi-person-circle icon"
          >
             
          </h1>
          <span
            class="caption"
          >
             
            Karen Rugerio Armenta
             
          </span>
        </div>
        <div
          class="locationSize"
        >
          <a
            class="caption2"
            href="https://itesm.zoom.us/j/5519309007"
          >
             
            https://itesm.zoom.us/j/5519309007
             
          </a>
        </div>
        <div
          class="statusSize"
        >
          <button
            class="container pending"
          >
             
            Pendiente
          </button>
        </div>
      </div>
      <div
        class="box"
      >
        <div
          class="dateSize"
        >
          <h1
            class="bi bi-alarm icon"
          >
             
          </h1>
          <span
            class="caption"
          >
             
            Lunes 27 de Marzo - 3:00 PM - 4:00 PM
             
          </span>
        </div>
        <div
          class="subjectSize"
        >
          <h1
            class="bi bi-stack icon"
          >
             
          </h1>
          <span
            class="caption"
          >
             
            Estructuras simbólicas en la imagen, la literatura y la música
             
          </span>
        </div>
        <div
          class="studentSize"
        >
          <h1
            class="bi bi-person-circle icon"
          >
             
          </h1>
          <span
            class="caption"
          >
             
            Karen Rugerio Armenta
             
          </span>
        </div>
        <div
          class="locationSize"
        >
          <a
            class="caption2"
            href="https://itesm.zoom.us/j/5519309007"
          >
             
            https://itesm.zoom.us/j/5519309007
             
          </a>
        </div>
        <div
          class="statusSize"
        >
          <button
            class="container pending"
          >
             
            Pendiente
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
